src/Actions/OutputsDslQueryAction.php

```diff
@@ -96,4 +110,32 @@ private function mapCondition(array $condition): array
 
         return $parser->parse($field, $value);
     }
-}
+
+    private function handleEnhancedDateValue(string $field, string $operator, array $value): array
+    {
+        if ($value['type'] === 'relative') {
+            if (isset($value['value'])) {
+                $relativeOperator = strtoupper($value['value']);
+
+                if (isset($this->operators[$relativeOperator])) {
+                    $parserClass = $this->operators[$relativeOperator];
+                    $parser = new $parserClass;
+                    return $parser->parse($field, ['operator' => $operator]);
+                }
+            } elseif (isset($value['base'], $value['offset'], $value['unit'])) {
+                $valueWithOperator = array_merge($value, ['operator' => $operator]);
+                $parser = new RelativeDateParser;
+                return $parser->parse($field, $valueWithOperator);
+            }
+        } elseif ($value['type'] === 'manual') {
+            $valueWithOperator = [
+                'date' => $value['value'],
+                'operator' => $operator
+            ];
+            $parser = new ManualDateParser;
+            return $parser->parse($field, $valueWithOperator);
+        }
+
+        return ['match_all' => []];
+    }
```

Comment:
I think all this logic should be moved to the parser, you should only need the field and value there as the parser itselfs would represent the operator